portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -27,7 +34,38 @@
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello Jacob!</h1>");
+    Query query = new Query("Task");
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    ArrayList<String> comments = new ArrayList<String>();
+    for (Entity entity : results.asIterable()) {
+      String comment = (String) entity.getProperty("comment");
+
+      comments.add(comment);
+    }
+
+    Gson gson = new Gson();
+
+    // Send the JSON as the response
+    response.setContentType("application/json;");
+    response.getWriter().println(gson.toJson(comments));
   }
-}
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the input from the form.
+    String text = request.getParameter("text-input");
+
+    // Add input to current comments in datastore.
+    Entity taskEntity = new Entity("Task");
+    taskEntity.setProperty("comment", text);
```

Comment:
Please use constants for these strings.

Reply:
Hey Tong, you'll see the changes in my next pull request